hpy/tools/autogen/public_api.h

```diff
@@ -90,7 +91,8 @@ HPy HPyErr_NoMemory(HPyContext ctx);
 
 /* object.h */
 int HPy_IsTrue(HPyContext ctx, HPy h);
-HPy HPyType_FromSpec(HPyContext ctx, HPyType_Spec *spec);
+HPy HPyType_FromSpec(HPyContext ctx, HPyType_Spec *spec,
+                     HPyType_SpecParam *optional_object_parameters);
```

Comment:
I'm not sure about the name `optional_object_parameters`:
  - `object`: see above the my comment for `void *` vs `HPy`: if we choose `void *`, they are not necessarily objects. If we choose `HPy`, it's "obvious" they are objects

 - `optional`: there are tons of params which accept `NULL`, we never put `optional` in the name

Maybe `params` is enough?